aten/src/ATen/native/TypeProperties.cpp

```diff
@@ -75,8 +78,14 @@ ResultTypeState update_result_type_state(const Tensor& tensor, const ResultTypeS
   }
   ResultTypeState new_state = in_state;
   ScalarType current = tensor.scalar_type();
-  if (tensor.unsafeGetTensorImpl()->is_wrapped_number() && isFloatingType(current)) {
-    current = typeMetaToScalarType(at::get_default_dtype());
+  if (tensor.unsafeGetTensorImpl()->is_wrapped_number()) {
+    auto current_default = typeMetaToScalarType(at::get_default_dtype());
+    if(isComplexType(current)) {
+      current = typeMetaToScalarType(at::get_default_complex_dtype());
+    }
+    if(isFloatingType(current)) {
```

Comment:
question: is it a conscious decision that a complex type is not also considered a floating type? I could see considering complex to be a special case of floating types.

Reply:
I also think complex should be considered a floating point type

Reply:
will make a follow-up PR for this

Reply:
let's update this to be `else if` anyway though so it's future proof to that change.